src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -553,12 +553,18 @@ private[internal] trait GlbLubs {
     val ntps: List[PolyType]   = ptHead :: ptRest.map(normalizeIter)
 
     val tparams1: List[Symbol] = {
-      def unifyBounds(ntp: PolyType): List[Type] = {
+      val substSymMaps: Array[SubstSymMap] = new Array[SubstSymMap](ptRest.length)
+      foreachWithIndex(ntps.tail) { (ntp, ix) =>
         val tparams1 = ntp.typeParams
-        tparams1 map (tparam => tparam.info.substSym(tparams1, tparamsHead))
+        if (! ((tparams1 eq tparamsHead) || tparams1.isEmpty))
+          substSymMaps(ix) = new SubstSymMap(tparams1, tparamsHead)
       }
-      val boundsTts : List[List[Type]] = ntps.tail.map(unifyBounds).transpose
-      map2(tparamsHead, boundsTts){ (tparam, bounds) =>
+      mapWithIndex(tparamsHead){ (tparam, ix) =>
```

Comment:
@retronym Thinking a bit more on this code, I have a doubt: when we use the functions `mapWithIndex` and `foreachWithIndex`, which are functions that take an `Int` as second parameter: does the execution of those loops perform a boxing/unboxing of that int value?

Reply:
They will, but that boxing is actually quite cheap because `Integer.valueOf` special cases small ints with a cache.